flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientHandlerTest.java

```diff
@@ -64,23 +75,36 @@
 	@Test(timeout = 60000)
 	@SuppressWarnings("unchecked")
 	public void testReleaseInputChannelDuringDecode() throws Exception {
-		// Mocks an input channel in a state as it was released during a decode.
-		final BufferProvider bufferProvider = mock(BufferProvider.class);
-		when(bufferProvider.requestBuffer()).thenReturn(null);
-		when(bufferProvider.isDestroyed()).thenReturn(true);
-		when(bufferProvider.addBufferListener(any(BufferListener.class))).thenReturn(false);
-
-		final RemoteInputChannel inputChannel = mock(RemoteInputChannel.class);
-		when(inputChannel.getInputChannelId()).thenReturn(new InputChannelID());
-		when(inputChannel.getBufferProvider()).thenReturn(bufferProvider);
-
-		final BufferResponse receivedBuffer = createBufferResponse(
-			TestBufferFactory.createBuffer(), 0, inputChannel.getInputChannelId(), 2);
-
-		final PartitionRequestClientHandler client = new PartitionRequestClientHandler();
-		client.addInputChannel(inputChannel);
-
-		client.channelRead(mock(ChannelHandlerContext.class), receivedBuffer);
+		final NetworkBufferPool networkBufferPool = new NetworkBufferPool(10, 32, MemoryType.HEAP);
```

Comment:
Thanks for those changes!

One more question, maybe it would be better to split refactoring of the old tests to separate commit? If this is too much work you can ignore this one, but it would make a cleaner history (and it would give you one more contribution ;) )

Reply:
Yes, I will split the olds tests in a separate commit later. :)